mysql/charset.go

```diff
@@ -15,8 +15,27 @@ package mysql
 
 import "unicode"
 
-// CharsetIDs maps charset name to its default collation ID.
-var CharsetIDs = map[string]uint8{
+// CharsetNameToID maps charset name to its default collation ID.
+func CharsetNameToID(charset string) uint8 {
+	// Use quick path for TiDB to avoid access charsetIDs map
+	// "SHOW CHARACTER SET;" to see all the supported character sets.
+	if charset == "utf8mb4" {
```

Comment:
Why use an if/else chain instead of an actual `switch`? I've benchmarked this locally using Go 1.12 and there's no observable difference (within ±0.05 ns/op).

Reply:
oh, the benchmark code i paste should `s/Swithch/If/`.
just want to make sure using linear search(if goto), if there's too many switch case it will be slow  
even if you `SwitchFirst`, because it's doing binary search on a sorted array with all case.